src/server/table_hotspot_policy.h

```diff
@@ -42,12 +44,59 @@ class hotspot_algo_qps_skew : public hotspot_policy
     }
 };
 
+// PauTa Criterion
+class hotspot_algo_qps_variance : public hotspot_policy
+{
+public:
+    void analysis(const std::queue<std::vector<hotspot_partition_data>> &hotspot_app_data,
+                  std::vector<::dsn::perf_counter_wrapper> &hot_points)
+    {
+        dassert(hotspot_app_data.back().size() == hot_points.size(),
+                "partition counts error, please check");
+        std::vector<double> data_samples;
+        data_samples.reserve(hotspot_app_data.size() * hot_points.size());
+        const auto &anly_data = hotspot_app_data.back();
+        auto temp_data = hotspot_app_data;
+        double avg = 0, sd = 0;
+        int n = 0;
+        // avg: Average number
+        // sd: Standard deviation
+        // n: Number of samples
+        while (!temp_data.empty()) {
+            for (auto partition_data : temp_data.front()) {
+                if (partition_data.total_qps - 1.00 > 0) {
+                    data_samples.push_back(partition_data.total_qps);
+                    avg += partition_data.total_qps;
+                    n++;
+                }
+            }
+            temp_data.pop();
+        }
+        if (n == 0) {
+            ddebug("hotspot_app_data size == 0");
+            return;
+        }
+        avg /= n;
+        for (auto data_sample : data_samples) {
+            sd += pow((data_sample - avg), 2);
+        }
+        sd = sqrt(sd / n);
+        for (int i = 0; i < hot_points.size(); i++) {
+            double hot_point = (anly_data[i].total_qps - avg) / sd;
+            // perf_counter->set can only be unsigned __int64
+            // use ceil to guarantee conversion results
+            hot_point = ceil(std::max(hot_point, double(0)));
+            hot_points[i]->set(hot_point);
+        }
+    }
+};
+
 // hotspot_calculator is used to find the hotspot in Pegasus
 class hotspot_calculator
 {
 public:
     hotspot_calculator(const std::string &app_name, const int partition_num)
-        : _app_name(app_name), _points(partition_num), _policy(new hotspot_algo_qps_skew())
+        : _app_name(app_name), _points(partition_num), _policy(new hotspot_algo_qps_variance())
```

Comment:
1. Read `hotspot_policy` name from config file
2. Create a `hotspot_policy` object according to the name
Then you can change hotspot_policy algorithm by update config, no need to recompile.